packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3994,7 +3995,7 @@ class InputDecoration {
   ///
   /// Only null valued properties from this [InputDecoration] are replaced
   /// by the corresponding values from [theme].
-  InputDecoration applyDefaults(InputDecorationTheme theme) {
+  InputDecoration applyDefaults(InputDecorationThemeData theme) {
```

Comment:
Also for the type of the parameter, I'm a little hesitant to change it as well. I feel it would be better if we can just keep minimum breaking changes in ThemeData